### PR TITLE
Multi cuisines fetching

### PR DESCRIPTION
**Pull request by tbarnahor**

### CODE REVIEW

File: project/src/main/java/com/google/sps/data/PlacesFetcher.java [hunk 1]
```diff
@@ -37,7 +37,7 @@
      * will be fields of a UserPrefrences instance passed to fetch() by the Servlet.
      */
     private static final LatLng LOCATION = new LatLng(32.080576, 34.780641); // Rabin Square TLV;
-    private static final String CUISINES = "sushi"; // TODO(M0): change to set of types;
+    private static final List<String> CUISINES_LIST = new ArrayList<>(List.of("sushi", "burger"));
```
Comment (arielz-g) [inline project/src/main/java/com/google/sps/data/PlacesFetcher.java]: Use an ImmutableList here (always prefer to use immutable types).

Comment (arielz-g) [inline project/src/main/java/com/google/sps/data/PlacesFetcher.java]: I'd just call it CUISINES, the _LIST is redundant as it's given from the object type.

File: project/src/main/java/com/google/sps/data/PlacesFetcher.java [hunk 2]
```diff
@@ -66,7 +66,7 @@
      */
     public ImmutableList<Place> fetch() throws IOException, InterruptedException, ApiException {
         TextSearchRequest query =
-            PlacesApi.textSearchQuery(CONTEXT, CUISINES, LOCATION)
+            PlacesApi.textSearchQuery(CONTEXT, genCuisinesStr(CUISINES_LIST), LOCATION)
```
Comment (arielz-g) [inline project/src/main/java/com/google/sps/data/PlacesFetcher.java]: You're passing a static class variable here. For now, as the list isn't an outside parameter, i'd just change the method to be static, and use the static list for now.

File: project/src/main/java/com/google/sps/data/PlacesFetcher.java [hunk 3]
```diff
@@ -137,4 +137,15 @@ PlaceDetails getPlaceDetails(PlaceDetailsRequest request)
             throws ApiException, InterruptedException, IOException {
         return request.await();
     }
+
+    private String genCuisinesStr(List<String> cuisineLst) {
```
Comment (arielz-g) [inline project/src/main/java/com/google/sps/data/PlacesFetcher.java]: This method name isn't that clear IMO. Always prefer to use explicit and understandable names both for methods and arguments (abbreviations aren't that common really).

Something like createCuisenesQuery sounds more fitting in this context.

File: project/src/main/java/com/google/sps/data/PlacesFetcher.java [hunk 4]
```diff
@@ -137,4 +137,15 @@ PlaceDetails getPlaceDetails(PlaceDetailsRequest request)
...
+        String cuisineStr = "";
```
Comment (arielz-g) [inline project/src/main/java/com/google/sps/data/PlacesFetcher.java]: The whole method can be simplified by staticlly importing:
import static java.util.stream.Collectors.joining;

and then:
return CUISINES.stream().collect(joining("|"));

Always try to search for a utility to do simple things for you:)

**Review by karenav — COMMENTED** [review]
Currently I don't have additional comments to what Ariel wrote :) 
The only thing I think is that we should use some sort of Enum for the cuisine types instead of strings. I'm not sure if it's in the scope of this PR or a different one, but I think it's something we should do.

**Comment by tbarnahor** [comment]
> Currently I don't have additional comments to what Ariel wrote :)
> The only thing I think is that we should use some sort of Enum for the cuisine types instead of strings. I'm not sure if it's in the scope of this PR or a different one, but I think it's something we should do.

See my comment on the UserPrefrence PR regarding this. I have an open issue about this and it will be done as part of M1 PRs.

File: project/src/main/java/com/google/sps/data/PlacesFetcher.java (hunk 1 above)
Comment (tbarnahor) [reply project/src/main/java/com/google/sps/data/PlacesFetcher.java]: Makes sense

Comment (tbarnahor) [reply project/src/main/java/com/google/sps/data/PlacesFetcher.java]: Done

File: project/src/main/java/com/google/sps/data/PlacesFetcher.java (hunk 3 above)
Comment (tbarnahor) [reply project/src/main/java/com/google/sps/data/PlacesFetcher.java]: I agree

File: project/src/main/java/com/google/sps/data/PlacesFetcher.java (hunk 4 above)
Comment (tbarnahor) [reply project/src/main/java/com/google/sps/data/PlacesFetcher.java]: Thanks, I don't naturally think about using streams and other java utilities, I will keep it in mind fot next times.

**Review by karenav — APPROVED** [review]

File: project/src/main/java/com/google/sps/data/PlacesFetcher.java [hunk 5]
```diff
@@ -137,4 +139,8 @@ PlaceDetails getPlaceDetails(PlaceDetailsRequest request)
             throws ApiException, InterruptedException, IOException {
         return request.await();
     }
+
+    private static String createCuisenesQuery() {
```
Comment (szwbar) [inline project/src/main/java/com/google/sps/data/PlacesFetcher.java]: typo: cuisines

File: project/src/main/java/com/google/sps/data/PlacesFetcher.java [hunk 6]
```diff
@@ -137,4 +139,8 @@ PlaceDetails getPlaceDetails(PlaceDetailsRequest request)
...
+        return CUISINES.stream().collect(joining("|"));
```
Comment (szwbar) [inline project/src/main/java/com/google/sps/data/PlacesFetcher.java]: Test?

Comment (szwbar) [reply project/src/main/java/com/google/sps/data/PlacesFetcher.java]: By the way, I think this can be simplified by calling
`String.join("|", CUISINES`);`

I don't see the benefit of using streams here.

Comment (tbarnahor) [reply project/src/main/java/com/google/sps/data/PlacesFetcher.java]: Yes the join works, thanks.

Regarding the testing, isn't it enough that it is tested as part of the fetch testing? Since CUISINES is hardcoded in this version we only need it to work for this specific list of cuisines without any corner cases, and so this is tested in the current tests as part of fetch testing.

**Review by arielz-g — APPROVED** [review]

File: project/src/main/java/com/google/sps/data/PlacesFetcher.java (hunk 6 above)
Comment (szwbar) [reply project/src/main/java/com/google/sps/data/PlacesFetcher.java]: The problem is that `getCuisinesQuery` is not tested now, and you'll probably not test it in the next PR (as it's an existing functionality). So the recommended way is to add the test directly when you add the functionality.
